### ISSUE TITLE
What is Time schedule difference? between torch.sigmoid(torch.randn((bs,)) and timesteps[random.randint(0, 999)

### ISSUE THREAD

**Issue by xilanhua12138** [XLabs-AI/x-flux#133]
In the file https://github.com/XLabs-AI/x-flux/commit/3139864620fb268eabfb3fe8f50141a963982840#diff-dcf224c4e9bbbf1401dce3c8338c64162c858fb21a9e8625e828637896adfcddR255
<img width="1355" alt="image" src="https://github.com/user-attachments/assets/718a2920-c5e2-4a3f-899d-449e62eb94db">

Why do you change:
t = torch.sigmoid(torch.randn((bs,), device=accelerator.device))
to:
t = torch.tensor([timesteps[random.randint(0, 999)]]).to(accelerator.device)

I found that in the first version, the training time is always near 0.5, and in the second version t has much wider span

**Comment by xilanhua12138** [comment]
@stazizov, @0xb1b1, @gradient-diffusion
